### ISSUE TITLE
hw05 ready for grading

### ISSUE THREAD

**Issue by arthursunbao** [arthursunbao/STAT545-hw01--Bao-Sun#6]

**Comment by arthursunbao** [comment]
@vincenzocoia @gvdr @ksedivyhaley @JoeyBernhardt @mynamedaike @pgonzaleze @derekcho

**Comment by estennw** [comment]
Hi @arthursunbao 

Great job! Here are some specific comments on your work:

### Question 1 Factores etc.
Nice drop of Oceania. Note that if there were 10000 continents it might would have been easier to do `h_result <- gapminder %>% filter(continent != "Oceania")`. Nice use of `fct_infreq()`, straight forward. If you want to order by something else than number of observations look at `fct_reorder()` from the same package. 

I missed plots showing the effect of reordering the data. I would encourage you to test reordering and not reordering the data before plotting, and observe the differences. When reordering, the arrangement of the factor variables change, as one would expect. 

### Question 2 Reading and writing to file
Good job! As you probably understood, writing to and reading from csv does not preserve factor ordering.

### Question 3 Remake figure
I do believe the intent with this question was to use some of the concepts that Tamara Munzner introduced in the guest lecture, and I don't believe that labeling the colors you used was one of those concepts. I agree that it might be helpful to label the points in the plot, but wouldn't it make more sense to use the country as the label? And I would advice you to not make the labels overlap. Concerning the colors, maybe it would be nice to use distinct colors to specify the continent? 

You save the figures to file correctly! Also try including them in the md file!

### Question 4 Repo cleanup
Nice done! Also consider naming the homeworks by the topics in the homework. And personally, I prefer adding one subfolder for md files, one for figures, one for tables etc. 


To summarize, good job! Hope you learned something from this class that you can use in other settings!

**Comment by shadowforti** [comment]
Hello 

Good job!

- The homework folder is well organized. I could easily find the homework
- The readme file clearly summarized the content
- Oceania was successfully dropped 
- Factor level was successfully removed 
- Factors were reordered perfectly
- The figure was improved by having colors and names
Some suggestions
- The plot picture somehow cannot be loaded
- It would be better if you have included a summary in the drop section to indicate that "Oceania" was successfully being dropped. 
- Maybe to include the arrange and reorder part would be a plus 
- Maybe you forgot to clean up and rename the folder? or I did not check correctly. 

Overall, the homework is great and I can see that you have learned a lot in the class. 

Thanks

Jian

**Comment by ksedivyhaley** [comment]
**Factor management (drop & reorder):** Partial (see comments)
**File I/O (data):** Yes (but see comments)
**Visualization design:** Partial (see comments)
**File I/O (write figure):** Partial (does not load saved file)
**Organized GitHub:** Partial (see comments)
**Bonus (more forcats, eg relevel):** No
**Reflection:** Yes

**Comments:**

- For your repo, add “slugs” to the files, eg hw05_FactorsFigures, so you know what is in each assignment – especially since you don't have this info in the table of contents. I also recommend using subfolders in each assignment to store saved files (hw05 is a bit crowded).
- The factor management section included a “common part” discussing the effect of arrange vs factor rearranging on a figure.
- Your exploration of data I/O is odd. You wouldn't usually chain the save functions in this way – instead we were expecting a comparison of the effects of each function, and perhaps comments on when you would want to use the different save formats.
- Similarly, the visualization design is not what we were looking for. First, the graphing is supposed to be done in ggplot, not base R. Second, the plot is supposed to demonstrate something about the gapminder data. Exploring colours and labels is nice, but the point of visualization design is to convey information in an easily understandable way.

Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.
